### PR TITLE
usb: Use enumerated return codes and usbd_control_complete_callback type

### PR DESCRIPTION
**Pull request by kuldeepdhaka**
fix examples for incomming PR libopencm3#497.

### CODE REVIEW

File: examples/stm32/f1/lisa-m-1/usb_dfu/usbdfu.c [hunk 1]
```diff
@@ -216,17 +218,17 @@ static int usbdfu_control_request(usbd_device *usbd_dev, struct usb_setup_data *
 		(*buf)[4] = usbdfu_state;
 		(*buf)[5] = 0; /* iString not used here */
 		*len = 6;
-		*complete = usbdfu_getstatus_complete;
-		return 1;
+		*complete = (usbd_control_complete_callback) usbdfu_getstatus_complete;
```
Comment (karlp) [inline examples/stm32/f1/lisa-m-1/usb_dfu/usbdfu.c]: why the cast? what type is missing here?

File: examples/stm32/f1/other/usb_cdcacm/cdcacm.c [hunk 2]
```diff
@@ -101,7 +101,7 @@ static const struct {
 		.bDescriptorType = CS_INTERFACE,
 		.bDescriptorSubtype = USB_CDC_TYPE_UNION,
 		.bControlInterface = 0,
-		.bSubordinateInterface0 = 1, 
+		.bSubordinateInterface0 = 1,
```
Comment (karlp) [inline examples/stm32/f1/other/usb_cdcacm/cdcacm.c]: please don't make unrelated whitespace changes, no matter how tempting it is.

Comment (kuldeepdhaka) [inline examples/stm32/f1/other/usb_cdcacm/cdcacm.c]: not intentional, (i have configured) editor automnatically does it.

**Comment by kuldeepdhaka** [comment]
@karlp removed all unrelated whitespace changes and complete callback casting

**Comment by karlp** [comment]
I still get warnings from a few examples with this one: 

* examples/stm32/f1/lisa-m-1/usb_dfu/
* examples/stm32/f1/lisa-m-1/usb_hid/
* examples/stm32/f1/other/usb_dfu/
* examples/stm32/f1/other/usb_hid/
* examples/stm32/f1/stm32-h103/usb_dfu/
* examples/stm32/f1/stm32-h103/usb_hid/
* examples/stm32/f1/stm32-h103/usb_iap/

Can you please check that you've got everything here?

**Comment by kuldeepdhaka** [comment]
https://github.com/libopencm3/libopencm3/pull/498 should be merged in libopencm3 before https://github.com/libopencm3/libopencm3/issues/497.
afaiu, the two commits prs are strongly coupled and the side effect is seen in form of warning.

**Comment by kuldeepdhaka** [comment]
no energy left to manage.
